client/src/Model/MainModel.ts

```diff
@@ -0,0 +1,36 @@
+import { dummyhistories } from '@/assets/dummy';
+import Observable from '@/Core/Observable';
+import { IHistory } from '@/utils/types';
+
+class MainModel extends Observable {
+  private historyCards: IHistory[];
+
+  constructor() {
+    super();
+    this.historyCards = [];
+    this.initHistoryCard();
+  }
+
+  initHistoryCard() {
+    dummyhistories.forEach((history) => {
+      const newHistory: IHistory = {
+        date: history.create_time,
+        type: history.type,
+        category: history.category,
+        content: history.content,
+        payment: history.payment,
+        price: history.price,
+      };
+      this.historyCards.push(newHistory);
+    });
+  }
+
+  addHistory(key: string, history: IHistory) {
```

Comment:
모델 내부에 `key` 값을 두면 모델 내부에서 데이터를 다루는 로직에서는

키 값을 따로 인자로 전달하지 않아도 될 거 같아요~~
```suggestion
  addHistory(history: IHistory) {
```

`Observable` 클래스에 있는 `subscribe`가 문제인데.. ㅠㅠ